src/core/tests/test_thread.py

```diff
@@ -3,11 +3,41 @@
 from threading import Thread
 
 def test01_use_scoped_thread_environment(variant_scalar_rgb):
-    def use_scoped_set_thred_env(env):
+    def use_scoped_set_thread_env(env):
         with mi.ScopedSetThreadEnvironment(environment):
             mi.Log(mi.LogLevel.Info, "Log from a thread environment.")
 
     environment = mi.ThreadEnvironment()
-    thread = Thread(target = use_scoped_set_thred_env, args = (environment, ))
+    thread = Thread(target = use_scoped_set_thread_env, args = (environment, ))
     thread.start()
     thread.join()
+
+def test02_access_logger_from_new_thread(variant_scalar_rgb):
+    def print_to_log():
+        mi.set_log_level(mi.LogLevel.Info)
+        mi.Log(mi.LogLevel.Info, "Log from a thread environment.")
+
+    thread = Thread(target = print_to_log)
+    thread.start()
+    thread.join()
```

Comment:
Let's add an assert to make sure that the log was visible (you can captue it with the `capsys` fixture).

Reply:
Good idea. I now added a `mi.StreamAppender`. I could not get it to work with `capsys`, this one only would capture Python-side prints for me (seems to be a known limitation)